### PR TITLE
GitHub name check

### LINKED ISSUE #128: Add validation to check if GitHub repo name already exists
As a user, I want tyr to check if the GitHub repo already exist. If it does, it should allow me to change it.

### PR DESCRIPTION
**Pull request by jackcmeyer**
closes #128 

* adds validation in cli.js to check if the name already exists on github as a repository, if it does, it will reprompt for a new project name. 
* adds relevant tests

### CODE REVIEW

File: src/cli.js [hunk 1]
```diff
@@ -155,5 +176,10 @@ export async function run(configFile, logFile) {
     return;
   }
 
+  if (configurations.toolingConfigurations.sourceControl === 'GitHub') {
+    configurations.projectConfigurations.projectName =
+      validateProjectNameAgainstGithubRepositories(configurations);
```
Comment (enclark) [inline src/cli.js]: We may need to await here.  Is this intentionally left out?

File: src/prompt/prompt.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ import chalk from 'chalk';
 
 import * as projectConfigurationValidator from '../utils/project-configuration-validator';
 import choices from '../constants/choices';
+import * as githubServie from '../services/github-service';
```
Comment (enclark) [inline src/prompt/prompt.js]: This is spelled wrong

File: src/services/github-service.js [hunk 3]
```diff
@@ -26,6 +26,49 @@ export async function isValidCredentials(username, password) {
   }
 }
 
+/**
+ * Gets the repositories for the given user
+ * @param username the username of the user
+ * @param password the password for the user
+ * @param pageNumber the page number of the get user repositories response response
```
Comment (enclark) [inline src/services/github-service.js]: pageNumber is not a param

File: src/services/github-service.js [hunk 4]
```diff
@@ -26,6 +26,49 @@ export async function isValidCredentials(username, password) {
...
+ */
+export function isValidGithubRepositoryName(repositoryName, repositories) {
+  let isValid = true;
+  repositories.forEach((repo) => {
+    if (repo.name === repositoryName) {
+      isValid = false;
+    }
+  });
+
+  return isValid;
```
Comment (enclark) [inline src/services/github-service.js]: Could use array.includes

File: src/cli.js (hunk 1 above)
Comment (jackcmeyer) [reply src/cli.js]: Yes. There is no need for an await here and await here because there is no asynchronous event.

Comment (jackcmeyer) [reply src/cli.js]: I take that back. An await should be here due to the prompt call.

File: src/prompt/prompt.js [hunk 5]
```diff
@@ -176,3 +177,24 @@ export async function promptForHerokuCredentials() {
...
 }
+
+export async function repromptForProjectName(repositories) {
+  console.log(chalk.blue('ReEnter a Valid Project Name: '));
+
+  const question = [{
+    name: 'projectName',
+    type: 'input',
+    message: 'Project Name:',
+    validate: (value) => {
```
Comment (KrashLeviathan) [inline src/prompt/prompt.js]: Will this validation block the user from continuing until they enter a valid name? If not, we'll need a `while` loop above.

Comment (jackcmeyer) [reply src/prompt/prompt.js]: Yes, this validation is built in to the prompt. Which prevents them from going forward until a name is valid. This is similar to the other prompt validations we have.

**Pull request merged**